app/assets/javascripts/assignments.js

```diff
@@ -250,10 +262,39 @@
     $(".file-picker").each(function(index) { activate_file_picker($(this)); });
   }
 
+  function toggles_init() {
+    $.each($('.toggle-wrapper'), function(index, toggleWrapper) {
+      $(toggleWrapper).click(function(e) {
+        e.stopPropagation(); // calling bootstrapToggle on the toggle itself fires the 'change' event, so the wrapper is necessary to change the toggle's value without firing the event
+        var toggle = $(toggleWrapper).find('.submission-enabled-toggle')[0];
+        var toggleId = toggle.id.split('-')[1];
```

Comment:
I might do this not as an id field, but as a data field, so that you're not munging and splitting strings...